lnwallet/aux_signer.go

```diff
@@ -0,0 +1,245 @@
+package lnwallet
+
+import (
+	"github.com/btcsuite/btcd/wire"
+	"github.com/lightningnetwork/lnd/fn"
+	"github.com/lightningnetwork/lnd/input"
+	"github.com/lightningnetwork/lnd/lntypes"
+	"github.com/lightningnetwork/lnd/lnwire"
+	"github.com/lightningnetwork/lnd/tlv"
+)
+
+// AuxHtlcDescriptor is a struct that contains the information needed to sign or
+// verify an HTLC for custom channels.
+type AuxHtlcDescriptor struct {
```

Comment:
Nice, with this change then we aren't as blocked on some proposed refactors as we've separated concerns by using a new minimal struct.